testhelper/certificatetest.go

```diff
@@ -36,66 +39,71 @@ type ECCertTuple struct {
 	PrivateKey *ecdsa.PrivateKey
 }
 
-// init runs before any other part of this package.
-func init() {
-	setupCertificates()
-}
-
 // GetRSARootCertificate returns root certificate signed using RSA algorithm
 func GetRSARootCertificate() RSACertTuple {
+	setupCertificates()
 	return rsaRoot
 }
 
 // GetRSALeafCertificate returns leaf certificate signed using RSA algorithm
 func GetRSALeafCertificate() RSACertTuple {
+	setupCertificates()
 	return rsaLeaf
 }
 
 // GetRSALeafCertificateWithoutEKU returns leaf certificate without EKU signed using RSA algorithm
 func GetRSALeafCertificateWithoutEKU () RSACertTuple {
-	return rsaLeaf
+	setupCertificates()
+	return rsaLeafWithoutEKU
 }
 
 // GetECRootCertificate returns root certificate signed using EC algorithm
 func GetECRootCertificate() ECCertTuple {
+	setupCertificates()
 	return ecdsaRoot
 }
 
 // GetECLeafCertificate returns leaf certificate signed using EC algorithm
 func GetECLeafCertificate() ECCertTuple {
+	setupCertificates()
 	return ecdsaLeaf
 }
 
 // GetUnsupportedRSACert returns certificate signed using RSA algorithm with key
 // size of 1024 bits which is not supported by notary.
 func GetUnsupportedRSACert() RSACertTuple {
+	setupCertificates()
 	return unsupportedRSARoot
 }
 
 // GetUnsupportedECCert returns certificate signed using EC algorithm with P-224
 // curve which is not supported by notary.
 func GetUnsupportedECCert() ECCertTuple {
+	setupCertificates()
 	return unsupportedECDSARoot
 }
 
 // GetRSASelfSignedSigningCertificate returns a self-signed certificate created which can be used for signing
 func GetRSASelfSignedSigningCertificate() RSACertTuple {
+	setupCertificates()
 	return rsaSelfSignedSigningCert
 }
 
 func setupCertificates() {
-	rsaRoot = getRSACertTuple("Notation Test RSA Root", nil)
-	rsaLeaf = getRSACertTuple("Notation Test RSA Leaf Cert", &rsaRoot)
-	rsaLeafWithoutEKU = getRSACertWithoutEKUTuple("Notation Test RSA Leaf without EKU Cert", &rsaRoot)
-	ecdsaRoot = getECCertTuple("Notation Test EC Root", nil)
-	ecdsaLeaf = getECCertTuple("Notation Test EC Leaf Cert", &ecdsaRoot)
-	unsupportedECDSARoot = getECCertTupleWithCurve("Notation Test Invalid ECDSA Cert", nil, elliptic.P224())
-
-	// This will be flagged by the static code analyzer as 'Use of a weak cryptographic key' but its intentional
-	// and is used only for testing.
-	k, _ := rsa.GenerateKey(rand.Reader, 1024)
-	unsupportedRSARoot = GetRSACertTupleWithPK(k, "Notation Unsupported Root", nil)
-	rsaSelfSignedSigningCert = GetRSASelfSignedSigningCertTuple("Notation Signing Test Root")
+	once.Do( func() {
```

Comment:
nit: we might need to run `go fmt` against this file.
```suggestion
	once.Do(func() {
```